src/atom-environment.coffee

```diff
@@ -229,6 +229,10 @@ class AtomEnvironment extends Model
 
     checkPortableHomeWritable()
 
+    # Patch the `process.env` on startup to fix the problem first documented
+    # in #4126
+    process.env = @project.getEnv()
```

Comment:
If you move the patching logic such that it is executed in the first atom-environment, then maybe `@project.getEnv()` could be renamed to be `@project.env`. Perhaps we could also externalize the logic into a file that is only required if patching is needed (`javascript` FTW?).